hoodie-hadoop-mr/src/main/java/com/uber/hoodie/hadoop/realtime/AbstractRealtimeRecordReader.java

```diff
@@ -109,6 +114,12 @@ public AbstractRealtimeRecordReader(HoodieRealtimeFileSplit split, JobConf job)
     }
   }
 
+  private boolean usesCustomPayload() {
+    HoodieTableMetaClient metaClient = new HoodieTableMetaClient(jobConf, split.getBasePath());
```

Comment:
Another option is to expose a method that returns boolean flag indicating merge logic is user controlled (i.e: read schema should contain all fields) ?

Reply:
Are you suggesting this config be set by the user ?

Reply:
Right,  More like a merge-policy enum : { PREFER_LATEST, PREFER_EARLIEST, CUSTOM }  . You can open a todo jira with "newbie" tag

Reply:
https://issues.apache.org/jira/browse/HUDI-155